### DISCUSSION TITLE
pinia-vite-hotUpdate-issue

### DISCUSSION THREAD

**Discussion by JiangBingLong** [vuejs/pinia#1722] [Help and Questions] [answered]
### Reproduction

pinia代码更新后，热重载输入的内容后，浏览器显示还是之前的

### Steps to reproduce the bug

1.使用官网模板
![1](https://user-images.githubusercontent.com/38178487/195665289-152f9851-a05b-4a8e-90ad-e81ac31b280c.png)
2.package.json版本记录
![2](https://user-images.githubusercontent.com/38178487/195665370-f91b982c-edd1-4f2d-8987-32afeebe4a51.png)
3.修改内容，保存运行程序
![3](https://user-images.githubusercontent.com/38178487/195665458-4912dc79-e654-4e60-8e34-24481e7b40fc.png)
![4](https://user-images.githubusercontent.com/38178487/195665499-333dfba2-57a1-4f97-8927-156a801a7b20.png)
4.再次修改内容，保存
![5](https://user-images.githubusercontent.com/38178487/195665539-514d2eee-1d58-4942-9e77-6283bd68365c.png)
![6](https://user-images.githubusercontent.com/38178487/195665549-3ffb566e-c060-419b-9fa2-5e77f18fc152.png)

### Expected behavior

更新app.ts文件 和 counter.ts 文件后。浏览器输出结果应该是最新的

### Actual behavior

浏览其输出结果还是之前的内容

### Additional information

希望查看！谢谢

**Comment by posva** [comment, marked as answer]
Search HMR in docs. There is a snippet to include **in every store file**.

I wish there was a way to apply the code only once but [it cannot be done, at least yet](https://github.com/vitejs/vite/issues/4500)

**Comment by JiangBingLong** [comment]
I tried, is ok！But,I have 10 stores. Is this code(import.meta.hot) added to every file(store)?

**Comment by leoadamo** [comment]
Hi guys! 

@posva, I'm having the same doubt just as @JiangBingLong mentioned. In case of having many stores, is that necessary to add the snippet to enable HMR in each one? I was wondering if there is any way to reuse the snippet by creating a function or something like that.

Also, it would be great if you could provide some example of enabling HMR in setup stores. Should I use the same syntax of options stores? 

Thank you in advance folks! 👊🏻
